lib/AstOfLlbc.ml

```diff
@@ -1494,9 +1513,20 @@ let is_str env var_id =
 
 let is_dst_var env var_id = Option.is_some (is_dst env (snd (lookup env var_id)))
 
+let is_box_place (p : C.place) =
+  match p.ty with
+  | C.TAdt (TBuiltin TBox, _) -> true
+  | _ -> false
+
 let expression_of_rvalue (env : env) (p : C.rvalue) : K.expr =
   match p with
   | Use op -> expression_of_operand env op
+  (* Generally, `&*p` == `p`, EXCEPT for when `p` is a `Box`, in which case, `*p` means to take its internal value, which is a compiler magic *)
+  | RvRef ({ kind = PlaceProjection (p, Deref); _},_)
```

Comment:
Does Eurydice represent `&T`, `&mut T`, `*const T` and `*mut T` the same? I think that's all the non-Box types that can have a ProjectionElem::Deref on them